### PR TITLE
Z-test

### PR DESCRIPTION
**Pull request by FBartos**
adds z-test to summary statistics + sensitivity analyses
fixes: https://github.com/jasp-stats/INTERNAL-jasp/issues/1828

### CODE REVIEW

**Comment by Kucharssim** [comment]
It seems that installing some jaspModules did not work which failed the tests. @vandenman, could you have a look please?

**Comment by vandenman** [comment]
Oh boy, this is going to be fun to fix...

1. jaspSummaryStatistics cannot be installed because
2. jaspFrequencies cannot be installed because
3. [conting is archived](https://cran.r-project.org/package=conting) because
4. [BMS is archived](https://cran.r-project.org/web/packages/BMS/index.html) because an email to the maintainer was undeliverable.

for now, I can set up a fork of `conting` and `BMS`, point the fork of `conting` to the fork of `BMS` and then point the jaspFrequencies to the fork of `conting`. That should allow the tests to run, but it's a horrible solution...

File: R/summarystatsbayesianztest.R
Comment (Kucharssim) [inline R/summarystatsbayesianztest.R:307]: 1. Why the underscore? Can you use something more informative?
2. Why do you use `tau` for prior standard deviation? (At least it seems to me that's what this argument is). This can be confusing because $\tau$ is commonly as a precision parameter.

File: R/summarystatsbayesianztest.R
Comment (Kucharssim) [inline R/summarystatsbayesianztest.R:323]: Put these calculations into functions as you reuse these formulas multiple times.

File: inst/Description.qml
Comment (Kucharssim) [inline inst/Description.qml:87]: use tabs

**Review by Kucharssim — CHANGES_REQUESTED** [review]
The code looks good, I have just a couple of comments. Also, please add unit tests and help documentation.

At this time, I cannot build jasp or install development modules, so I cannot test this in JASP (I tested it somewhat in R and it seems to work). Perhaps @vandenman could install and briefly test? Otherwise we could merge this after the comments are resolved and test it with a nightlie later.

File: R/summarystatsbayesianztest.R
Comment (vandenman) [reply R/summarystatsbayesianztest.R:263]: [`metR`](https://github.com/eliocamp/metR) also adds a bootload of dependencies (See https://github.com/eliocamp/metR/blob/master/DESCRIPTION). @FBartos is it really important to show the numbers on top of the contour lines?

Comment (FBartos) [reply R/summarystatsbayesianztest.R:263]: they look really nice though :(

I was looking whether I could copy some of the functions to cut on dependencies, but it would be 300+ lines of code at least.... :/

**Comment by FBartos** [comment]
ok, now the installation does not even work for me :(

File: R/summarystatsbayesianztest.R
Comment (vandenman) [reply R/summarystatsbayesianztest.R:263]: Yeah, I just tried and it's a ton of code (I gave up after 500+ lines of code). All the more reason not to do this IMO. If you really want this though, I suggest you cut it out of metR and make a PR to jaspGraphs.

File: R/summarystatsbayesianztest.R
Comment (Kucharssim) [inline R/summarystatsbayesianztest.R:115]: Spotted by @vandenman: You can (should) use `jaspBase::.recodeBFtype()` for flipping the Bayes factors here

**Comment by Kucharssim** [comment]
Seems like the tests failed because of GitHub rate limit. @vandenman, I thought that's supposed to be fixed by using GITHUB_PAT in the workflow, right? Any idea why it fails here?

**Comment by vandenman** [comment]
@Kucharssim this is the first time I've seen it hit the rate limit. The GITHUB_PAT is set [here](https://github.com/jasp-stats/jasp-actions/blob/15882fca355696c7b5a34a7837e64b9cf938592e/.github/workflows/unittests.yml#L57) but for some reason, we've hit the limit of 1000/hour ([ref](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#requests-from-github-actions)). I thought that the GITHUB_PAT wasn't shared across repositories, but I'm not 100% sure. Another reason could be that jaspSummaryStatistics has too many GitHub dependencies.

File: R/summarystatsbayesianztest.R
Comment (FBartos) [reply R/summarystatsbayesianztest.R]: ok, I removed it for now :(

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:71]: tabs

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:88]: tabs

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:98]: tabs

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:104]: tabs

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:136]: tabs

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:228]:
```suggestion
					label:			qsTr("Prior std    min:")
```
don't mix tabs and spaces

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:238]:
```suggestion
					label:			qsTr("max")
```
don't mix tabs and spaces

File: inst/qml/SummaryStatsBayesianZTest.qml
Comment (vandenman) [inline inst/qml/SummaryStatsBayesianZTest.qml:257]: tabs

File: R/summarystatsbayesianztest.R
Comment (vandenman) [inline R/summarystatsbayesianztest.R:51]:
```suggestion
  options[["dataEs"]]  <- jaspBase::.parseRCodeInOptions(options[["dataEs"]])
  options[["dataSe"]]  <- jaspBase::.parseRCodeInOptions(options[["dataSe"]])
  options[["dataLCi"]] <- jaspBase::.parseRCodeInOptions(options[["dataLCi"]])
  options[["dataUCi"]] <- jaspBase::.parseRCodeInOptions(options[["dataUCi"]])

  options[["priorMean"]] <- jaspBase::.parseRCodeInOptions(options[["priorMean"]])
  options[["priorSd"]]   <- jaspBase::.parseRCodeInOptions(options[["priorSd"]])
```
`.parseRCodeInOptions` is [exported](https://github.com/jasp-stats/jaspBase/blob/dec4a6fef22a9ee75688a65bdff0c7cb9f7f209c/NAMESPACE#L28) so you only need `::`.

File: R/summarystatsbayesianztest.R
Comment (vandenman) [inline R/summarystatsbayesianztest.R:175]: how about
```suggestion
    BF         = if(options[["plotPriorAndPosteriorAdditionalInfo"]] && options[["bayesFactorType"]] == "BF01") { 1/BF10 else BF10},
    bfType     = if(options[["plotPriorAndPosteriorAdditionalInfo"]] && options[["bayesFactorType"]] == "BF01") { "BF01" else "BF10"},
```
?

File: R/summarystatsbayesianztest.R
Comment (vandenman) [inline R/summarystatsbayesianztest.R:203]:
```suggestion
    bfBreaks <- vapply(bfBreaks, FUN = function(bfBreak) eval(parse(text = bfBreak)), FUN.VALUE = numeric(1))
```

**Review by vandenman — CHANGES_REQUESTED** [review]
Looks good to me! Below are some pedantic comments.

File: R/summarystatsbayesianztest.R
Comment (FBartos) [reply R/summarystatsbayesianztest.R:203]: any specific reason for using vapply there?

**Comment by vandenman** [comment]
### About `Specify Bayes factor contours`
- [ ] Throws an error for invalid input:
image

- [ ] The QML option doesn't look nice:
image

I need to zoom in to 160% to get this:
image

:/

- [ ] In the name `Specify Bayes factor countours` here `countours` should be `contours`.



### The prior-posterior plot:
image

- [ ] the plot has an incorrect title.
- [ ] the plot has incorrect default dimensions (see the t-tests for those).

File: R/summarystatsbayesianztest.R
Comment (vandenman) [reply R/summarystatsbayesianztest.R:203]: yes, always use `vapply` (or something from `purrr`) over `sapply` so that if it fails it fails early (defensive programming). But see the bug below, you're parsing user input as code without validation, which is a bit tricky. I wonder if there is a better way to do this.

Comment (vandenman) [reply R/summarystatsbayesianztest.R:203]: In fact, `sapply` is set to become one of the unwanted functions once we have [lintr](https://github.com/r-lib/lintr) set up.

File: R/summarystatsbayesianztest.R
Comment (FBartos) [reply R/summarystatsbayesianztest.R:175]: I think that those two are actually different.

My returns null when ``options[["plotPriorAndPosteriorAdditionalInfo"]]`` is FALSE and selects one of the remaining options based on ``options[["bayesFactorType"]]``

Yours does something oddly different? (also the ``else`` within the curly brackets?)

**Comment by FBartos** [comment]
I think I fixed everything but the ``Specify Bayes factor contours`` box input -- any idea how to put the input field below the text?

Unfortunately, I cannot test the code since the module won't install... :/

File: R/summarystatsbayesianztest.R
Comment (vandenman) [reply R/summarystatsbayesianztest.R:182]: > My returns null when options[["plotPriorAndPosteriorAdditionalInfo"]] is FALSE

Ah you're right, I missed that.

**Comment by vandenman** [comment]
So there is something weird going on. if you switch to `Effect size & CI (log)` you get this:
image

that is probably not the intended output.

If you click prior and posterior you get this error message:
image

once I enter the `Effect size log` field and press enter without changing the value, and do the same for `Confidence interval log` - `lower`, then I do get normal output and the plot does not error... :thinking:

The rest looks all fine to me.

**Comment by FBartos** [comment]
Oh, I remember this bug -- the double field is dependent on the input but changing the dropdown only changes the double field in qml, not R. I think that @boutinb knows to fix that.

**Review by vandenman — APPROVED** [review]
Unless @Kucharssim has any other comments this can be merged IMO.

**Review by Kucharssim — APPROVED** [review]

**Pull request merged**
